cmd/plume/release.go

```diff
@@ -518,25 +518,62 @@ func modifyReleaseMetadataIndex(spec *fcosChannelSpec, commitId string) {
 		plog.Fatalf("unmarshaling release metadata json: %v", err)
 	}
 
-	url, err := url.Parse(fmt.Sprintf("https://%s.s3.amazonaws.com/prod/streams/%s/builds/%s/release.json", spec.Bucket, specChannel, specVersion))
+	releasePath := filepath.Join("prod", "streams", specChannel, "builds", specVersion, "release.json")
+	url, err := url.Parse(fmt.Sprintf("https://%s.s3.amazonaws.com/%s", spec.Bucket, releasePath))
 	if err != nil {
 		plog.Fatalf("creating metadata url: %v", err)
 	}
 
+	releaseFile, err := api.DownloadFile(spec.Bucket, releasePath)
+	if err != nil {
+		plog.Fatalf("downloading release metadata: %v", err)
+	}
+	defer releaseFile.Close()
+
+	releaseData, err := ioutil.ReadAll(releaseFile)
+	if err != nil {
+		plog.Fatalf("reading release metadata: %v", err)
+	}
+
+	var im IndividualReleaseMetadata
+	err = json.Unmarshal(releaseData, &im)
+	if err != nil {
+		plog.Fatalf("unmarshaling release metadata: %v", err)
+	}
+
+	var commits []Commit
+	for arch, vals := range im.Architectures {
+		commits = append(commits, Commit{
+			Architecture: arch,
+			Checksum:     vals.Commit,
+		})
+	}
+
 	newRel := BuildMetadata{
-		CommitHash: specCommitId,
+		CommitHash: commits,
 		Version:    specVersion,
 		Endpoint:   url.String(),
 	}
 
 	for i, rel := range m.Releases {
-		if rel == newRel {
+		if compareStaticReleaseInfo(rel, newRel) {
 			if i != (len(m.Releases) - 1) {
 				plog.Fatalf("build is already present and is not the latest release")
```

Comment:
Was this supposed to disappear too?

Reply:
This is intended imo. I believe this falls into a case that Plume should not be handling which is essentially the case of republishing an older build as the most recent. In scenarios where that needs to occur it should be a deliberate manual action.